### PR TITLE
Add base structure for the MISO element

### PR DESCRIPTION
**Pull request by emmadrigal**

### CODE REVIEW

**Comment by emmadrigal** [comment]
This PR includes:
* the base TIOVX Mosi structure
* Downstream allocation
* Basic testing for pushing buffers

File: gst-libs/gst/tiovx/gsttiovxmiso.c [hunk 1]
```diff
@@ -0,0 +1,355 @@
...
+  G_OBJECT_CLASS (gst_tiovx_miso_parent_class)->finalize (obj);
+}
+
+static GstFlowReturn
+gst_tiovx_miso_aggregate (GstAggregator * aggregator, gboolean timeout)
+{
+  GstTIOVXMiso *self = GST_TIOVX_MISO (aggregator);
+  GstBuffer *outbuf = NULL;
+  GstFlowReturn ret = GST_FLOW_ERROR;
+  GList *l;
```
Comment (jafet-chaves) [inline gst-libs/gst/tiovx/gsttiovxmiso.c:249]: Initialize

File: gst-libs/gst/tiovx/gsttiovxmiso.c [hunk 2]
```diff
@@ -0,0 +1,355 @@
...
+      gst_query_remove_nth_allocation_pool (query, npool);
+    }
+
+    gst_object_unref (pool);
+  }
+
+  if (pool_needed) {
+    /* Temporary code to create a valid exemplar, this responsability will be
+       passed to the child class */
+    GstCaps *caps = NULL;
```
Comment (jafet-chaves) [inline gst-libs/gst/tiovx/gsttiovxmiso.c:335]: Do you need to call unref on this GstCaps?

**Review by jafet-chaves — CHANGES_REQUESTED** [review]

File: gst-libs/gst/tiovx/gsttiovxmiso.c (hunk 2 above)
Comment (emmadrigal) [reply gst-libs/gst/tiovx/gsttiovxmiso.c:335]: `query_parse_allocation` is transfer none: https://gstreamer.freedesktop.org/documentation/gstreamer/gstquery.html?gi-language=c#gst_query_parse_allocation

File: gst-libs/gst/tiovx/gsttiovxmiso.c [hunk 3]
```diff
@@ -0,0 +1,390 @@
...
+
+#ifdef HAVE_CONFIG_H
+#  include "config.h"
+#endif
+
+#include "gsttiovxmiso.h"
+
+#include "gsttiovxbufferpool.h"
+#include "gsttiovxcontext.h"
+#include "gsttiovxmeta.h"
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/tiovx/gsttiovxmiso.c:70]: This one is not used here

File: gst-libs/gst/tiovx/gsttiovxmiso.c [hunk 4]
```diff
@@ -0,0 +1,390 @@
...
+#include "gsttiovxutils.h"
+
+#include <app_init.h>
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/tiovx/gsttiovxmiso.c:73]: Is this used here?

File: gst-libs/gst/tiovx/gsttiovxmiso.c [hunk 5]
```diff
@@ -0,0 +1,390 @@
...
+{
+  GstVideoInfo in_info;
+  GstVideoInfo out_info;
+  GstTIOVXContext *tiovx_context;
+  vx_context context;
+  vx_graph graph;
+  vx_node node;
+  GList *input;
+  vx_reference *output;
+  guint out_pool_size;
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/tiovx/gsttiovxmiso.c:100]: Can we handle the output pool size as the input pads? as part of the GstTIOVXPad

File: gst-libs/gst/tiovx/gsttiovxmiso.c [hunk 6]
```diff
@@ -0,0 +1,390 @@
...
+typedef struct _GstTIOVXMisoPrivate
+{
+  GstVideoInfo in_info;
+  GstVideoInfo out_info;
+  GstTIOVXContext *tiovx_context;
+  vx_context context;
+  vx_graph graph;
+  vx_node node;
+  GList *input;
+  vx_reference *output;
```
Comment (rrcarlosrodriguez) [inline gst-libs/gst/tiovx/gsttiovxmiso.c:99]: Why don't we use GstTIOVXPad for the output as we do for the input?

**Review by rrcarlosrodriguez — CHANGES_REQUESTED** [review]

File: gst-libs/gst/tiovx/gsttiovxmiso.c (hunk 5 above)
Comment (emmadrigal) [reply gst-libs/gst/tiovx/gsttiovxmiso.c:100]: We aren't using TIOVXPad;s here, but in the next PR this will be moved to a MISO pad

**Pull request merged**